### PR TITLE
[Parse] Lexer build backtick trivia around espaced identifier token

### PR DESCRIPTION
**Pull request by omochi**
There was a logic that build backtick trivia before and after of escaped identifier token at two places.
First is at `swift::tokenizeWithTrivia` in `Parser.cpp`.
Second is at `SyntaxParsingContext::addToken` in `SyntaxParsingContext.cpp`.

This PR moves this logic into Lexer and makes them common.

First commit is adding of testcase for existing.
To check first case, it runs `swift-syntax-test -dump-full-tokens`.
To check second case, it runs `swift-syntax-test -serialize-raw-tree`.

Second commit is actual patch.

### CODE REVIEW

File: test/Syntax/tokens_escaped_identifier.swift [hunk 1]
```diff
@@ -0,0 +1,4 @@
+// RUN: %swift-syntax-test -input-source-filename %s -dump-full-tokens > %t
```
Comment (rintaro) [inline test/Syntax/tokens_escaped_identifier.swift]: Essentially, `-dump-full-tokens` is for debug use only.
So could you change this to `FileCheck` based test?
like:
```
// CHECK-LABEL 3:25
// CHECK-NEXT: (token identifier
// CHECK-NEXT:  (trivia block_comment
// CHECK-NEXT:  (trivia space 1)
// CHECK-NEXT:  (trivia backtick 1)
// CHECK-NEXT:  (text="if")
// CHECK-NEXT:  (trivia backtick 1)

// CHECK-LABEL 4:27
// CHECK-NEXT: (token identifier
// CHECK-NEXT: ...
```

**Review by rintaro — COMMENTED** [review]
Change itself looks good 👍
Comments for test cases.

File: test/Syntax/serialize_escaped_identifier.swift [hunk 2]
```diff
@@ -0,0 +1,4 @@
+// RUN: %swift-syntax-test -input-source-filename %s -serialize-raw-tree > %t
```
[the remark this answers is not in the document]
Comment (omochi) [reply test/Syntax/serialize_escaped_identifier.swift]: Thanks review.
Ah I have also felt that this test is fragile .
Ok, I will remove this.

File: test/Syntax/tokens_escaped_identifier.swift (hunk 1 above)
Comment (omochi) [reply test/Syntax/tokens_escaped_identifier.swift]: OK, I will try it.

**Comment by omochi** [comment]
I updated commits. Please check it.

**Comment by rintaro** [comment]
@swift-ci  Please test

**Comment by rintaro** [comment]
@swift-ci Please test

**Comment by omochi** [comment]
thank you

**Pull request merged**